### PR TITLE
added cs5 support in read/write.magpie

### PR DESCRIPTION
**Pull request by tscheypidi**
added new csv-variant "cs5" which is pretty similar but more generalized compared to "cs4" (left cs4 untouched to avoid backwards compatibility issues)

### CODE REVIEW

File: R/read.magpie.R [hunk 1]
```diff
@@ -130,11 +131,31 @@ read.magpie <- function(file_name, file_folder = "", file_type = NULL, as.array
...
                   check.names = check.names, stringsAsFactors = TRUE)
     readMagpie <- as.magpie(x, tidy = TRUE)
     attr(readMagpie, "comment") <- .readComment(fileName, commentChar = comment.char)
+  } else if (fileType == "cs5") {
+    .metaExtract <- function(fileName, commentChar) {
+      comment <- .readComment(fileName, commentChar = commentChar)
+      m <- grep("^META ", comment)
+      metadata <- comment[m]
+      comment <- comment[-m]
+      mNames <- sub("^META (.*):.*$", "\\1", metadata)
```
Comment (pfuehrlich-pik) [inline R/read.magpie.R:145]: This breaks when `:` is in one of the names, e.g. `*META names: x, y, coun:try, cell`. This is admittedly very unlikely, but it's easy to fix:
```suggestion
      mNames <- sub("^META ([^:]*):.*$", "\\1", metadata)
```

File: R/read.magpie.R [hunk 2]
```diff
@@ -130,11 +131,31 @@ read.magpie <- function(file_name, file_folder = "", file_type = NULL, as.array
...
+      mData <- strsplit(sub("^META .*: (.*)$", "\\1", metadata), ", ")
```
Comment (pfuehrlich-pik) [inline R/read.magpie.R:146]:
```suggestion
      mData <- strsplit(sub("^META [^:]*: (.*)$", "\\1", metadata), ", ")
```

File: R/write.magpie.R [hunk 3]
```diff
@@ -310,6 +310,22 @@ write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, appen
...
+          out <- c(out, paste0(commentChar, "META ", n, ": ", paste(meta[[n]], collapse = ", ")))
+        }
+        return(out)
+      }
+
+      zz <- file(filePath, open = "w")
+      if (any(comment != "")) writeLines(paste(comment.char, comment, sep = ""), zz)
+      writeLines(.formatMeta(attributes(output)[c("names", "dimtype")], comment.char), zz)
+      write.table(output, file = zz, quote = FALSE, row.names = FALSE, col.names = FALSE, sep = ",")
+      close(zz)
```
Comment (pfuehrlich-pik) [inline R/write.magpie.R:328]:
```suggestion
```
Not necessary when using withr when opening the file.

File: R/write.magpie.R [hunk 4]
```diff
@@ -310,6 +310,22 @@ write.magpie <- function(x, file_name, file_folder = "", file_type = NULL, appen
...
+
+      .formatMeta <- function(meta, commentChar) {
+        out <- NULL
+        for (n in names(meta)) {
+          out <- c(out, paste0(commentChar, "META ", n, ": ", paste(meta[[n]], collapse = ", ")))
+        }
+        return(out)
+      }
+
+      zz <- file(filePath, open = "w")
```
[the remark this answers is not in the document]
Comment (tscheypidi) [reply R/write.magpie.R:324]: as `withr` is currently only a suggested package but not imported I stick to the admittedly slightly inferior formulation

File: R/read.magpie.R (hunk 1 above)
Comment (tscheypidi) [reply R/read.magpie.R]: fixed

**Pull request merged**